src/payloads/Attendance.ts

```diff
@@ -1,24 +1,22 @@
-import { ValidateNested, IsOptional, IsBoolean, IsNumber, IsInstance } from 'class-validator';
-import { AppUser, Event } from '@Entities';
+import { IsOptional, IsBoolean, IsNumber, IsInstance } from 'class-validator';
+import { AppUserEventResponse } from './AppUser';
+import { EventAttendanceResponse } from './Event';
 
 abstract class BaseAttendancePayload {
-  @ValidateNested({ each: true })
-  @IsInstance(AppUser)
-  readonly attendee: AppUser;
+  @IsInstance(AppUserEventResponse)
+  readonly attendee: AppUserEventResponse;
 
-  @ValidateNested({ each: true })
-  @IsInstance(Event)
-  readonly event: Event;
+  @IsInstance(EventAttendanceResponse)
```

Comment:
i don't think this is good because the endpoint is /events/:eventID/signin, so event shouldn't be in the payload for the request. it's fine if it's in the response but it it's not in the request then it shouldn't be in the BasePayload

Reply:
Yep agreed. I removed BaseAttendancePayload completely and just have AttendanceResponse by itself now